parser.y

```diff
@@ -6034,32 +6038,118 @@ TableOptimizerHintList:
 	{
 		$$ = append($1.([]*ast.TableOptimizerHint), $2.(*ast.TableOptimizerHint))
 	}
+|	TableOptimizerHintList ',' TableOptimizerHintOpt
+	{
+		$$ = append($1.([]*ast.TableOptimizerHint), $3.(*ast.TableOptimizerHint))
+	}
 
 TableOptimizerHintOpt:
-	tidbSMJ '(' HintTableList ')'
+	hintSMJ '(' HintTableAndIndexList ')'
 	{
 		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1), Tables: $3.([]model.CIStr)}
 	}
-|	tidbINLJ '(' HintTableList ')'
+|	hintINLJ '(' HintTableAndIndexList ')'
 	{
 		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1), Tables: $3.([]model.CIStr)}
 	}
-|	tidbHJ '(' HintTableList ')'
+|	hintHJ '(' HintTableAndIndexList ')'
 	{
 		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1), Tables: $3.([]model.CIStr)}
 	}
-|	tidbHASHAGG '(' ')'
+|	hintUseIndexMerge '(' HintTableAndIndexList ')'
 	{
-		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1)}
+		$$ = &ast.TableOptimizerHint{
+			HintName: model.NewCIStr($1),
+			Tables:   $3.([]model.CIStr)[:1],
+			Indexes:  $3.([]model.CIStr)[1:],
+		}
 	}
-|	tidbSTREAMAGG '(' ')'
+|	hintUseToja '(' HintTrueOrFalse ')'
 	{
-		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1)}
+		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1), HintFlag: $3.(bool)}
+	}
+|	hintEnablePlanCache '(' HintTrueOrFalse ')'
+	{
+		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1), HintFlag: $3.(bool)}
 	}
 |	maxExecutionTime '(' NUM ')'
 	{
 		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1), MaxExecutionTime: getUint64FromNUM($3)}
 	}
+|	hintUsePlanCache '(' ')'
+	{
+		// arguments not decided yet.
+		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1)}
+	}
+|	hintQueryType '(' HintQueryType ')'
+	{
+		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1), QueryType: model.NewCIStr($3.(string))}
+	}
+|	hintMemoryQuota '(' HintMemoryQuota ')'
+	{
+		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1), MemoryQuota: $3.(uint64)}
+	}
+|	hintHASHAGG
+	{
+		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1)}
+	}
+|	hintSTREAMAGG
+	{
+		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1)}
+	}
+|	hintNoIndexMerge
+	{
+		$$ = &ast.TableOptimizerHint{HintName: model.NewCIStr($1)}
+	}
+|	hintReadConsistentReplica
```

Comment:
Is this one enforce table read from follower or learner?

Reply:
Does it also need a table list?

Reply:
This is depend on TiKV. It  will not be used in 4.0GA, so it's just a draft.

Reply:
It's about read from follower, but not decided to enforce or advise.